DataProcessing/CryptoCoarseFundamentalUniverseDataDownloader.cs

```diff
@@ -0,0 +1,283 @@
+                    {
+                        foreach (ZipArchiveEntry entry in zip.Entries)
+                        {
+                            string name = entry.Name.Split(".").First().ToUpper();
+
+                            var streamFile = entry.Open();
+                            var lines = ReadLines(streamFile);
+
+                            foreach (var line in lines)
+                            {
+                                var csv = line.Split(",");
+                                var dateTime = Parse.DateTimeExact(csv[0].Split(" ").First(), "yyyyMMdd");
+                                if (dateTime < _fromDate || dateTime > _toDate)
+                                {
+                                    continue;
+                                }
+
+                                var date = $"{dateTime:yyyyMMdd}";
+                                var high = decimal.Parse(csv[2], NumberStyles.Any, CultureInfo.InvariantCulture);
+                                var low = decimal.Parse(csv[3], NumberStyles.Any, CultureInfo.InvariantCulture);
+                                var price = decimal.Parse(csv[4], NumberStyles.Any, CultureInfo.InvariantCulture);
+                                var volume = decimal.Parse(csv[5], NumberStyles.Any, CultureInfo.InvariantCulture);
+                                var dollarVolume = price * volume;
+
+                                var content = new Dictionary<string, string>();
+                                dataByDate.TryAdd(date, content);
+                                if (dataByDate[date].TryGetValue(name, out var tempData))
+                                {
+                                    var temp = tempData.Split(",");
+                                    var oldVolume = decimal.Parse(temp[1], NumberStyles.Any, CultureInfo.InvariantCulture);
+                                    var oldDollarVolume = decimal.Parse(temp[2], NumberStyles.Any, CultureInfo.InvariantCulture);
+                                    var oldHigh = decimal.Parse(temp[4], NumberStyles.Any, CultureInfo.InvariantCulture);
+                                    var oldLow = decimal.Parse(temp[5], NumberStyles.Any, CultureInfo.InvariantCulture);
+
+                                    var newVolume = oldVolume + volume;
+                                    var newDollarVolume = oldDollarVolume + newVolume * price;
+                                    var newHigh = Math.Max(oldHigh, high);
+                                    var newLow = Math.Min(oldLow, low);
+
+                                    dataByDate[date][name] = $"{price},{newVolume},{newDollarVolume},{temp[3]},{newHigh},{newLow},";
+                                }
+                                else
+                                {
+                                    dataByDate[date][name] = $"{price},{volume},{dollarVolume},{csv[1]},{high},{low},";
```

Comment:
We can remove the last `,` been written?

Reply:
Hi Martin! That was left there intentionally, check line 216, a new attribute will be added at the end